js/particle-modeler/index.js

```diff
@@ -111,9 +118,35 @@ export default class Interactive extends PureComponent {
         newState["showNewAtom"+el] = false;
         _this.setState(newState);
         _this.addNewDraggableAtom(el);
+      } else {
+        if (d.x > 4.187 && d.x < 4.328 && d.y > 0.141 && d.y < 0.287) {
+          // mark atoms for deletion
+          if (!d.marked) {
+            _this.setState({deleteHover: true});
+            api.setAtomProperties(i, {marked: 1});
+          }
+        } else if (d.marked) {
+            _this.setState({deleteHover: false});
+          api.setAtomProperties(i, {marked: 0});
+        }
       }
     });
 
+    let deleteMarkedAtoms = function() {
+      let atomsToDelete = [];
+      for (let i=0, ii=api.getNumberOfAtoms(); i<ii; i++) {
+        if (api.getAtomProperties(i).marked)
+          atomsToDelete.push(i);
+      }
+      for (let i=atomsToDelete.length-1; i>-1; i--) {
+        api.removeAtom(atomsToDelete[i]);
+      }
+
+      _this.setState({deleteHover: false});
```

Comment:
Random comment: `let _this = this;` can be now replaced by arrow functions `=> { ... }` that are part of the ES6 standard. Of course it's matter of taste, but just saying in case you've missed this ES6 feature.